### ISSUE TITLE
Vulkan is extremely unstable

### ISSUE THREAD

**Issue by Vichy97** [maplibre/maplibre-native#3092]
### MapLibre Android Version

11.7.0

### Android SDK Version

Android 13

### Device

Google Pixel 8 and simulator

### What happened?

When using the Vulkan version of MapLibre, my app crashes constantly. Every 10 seconds or so even if I'm not on the screen with a map. The stack traces are very short but all are SIGSEGV crashes. 

```
Crashed: null pointer dereference
SIGSEGV 0x0000000000000064
```

### Steps to reproduce

Use Vulkan version of MapLibre

### Relevant log output

```shell

```

**Comment by louwers** [comment]
Without reproduction steps we're unable to investigate the issue you are facing.

Please provide more details.

**Comment by Vichy97** [comment]
Theres not really any more details to give. The logs only show what I posted. They don't give any kind of stacktrace.

**Comment by louwers** [comment]
@Vichy97 Are you using a particular style?

Are you using a particular feature or API?

Can you try it with a simple demo app and share that?

**Comment by Vichy97** [comment]
It seems to happen with any style. Tried several. My map is using the location api and the scalebar plugin, but nothing unusual other than that. I am running it inside `AndroidView` within Compose. I'll try to make a minimal repro example.

**Comment by louwers** [comment]
@Vichy97 Sounds like an issue with the scalebar plugin, not with Vulkan.

**Comment by Vichy97** [comment]
Maybe but it's working fine without vulkan. I thought scalebar just rendered a normal android view based on output from the map. I would guess that's not causing native code crashes.

**Comment by louwers** [comment]
Yes looks like it is not doing anything OpenGL ES related. But maybe it is hooking in some kind of callback that is not yet implemented for Vulkan yet.

Let's confirm it's indeed the plugin. Then we can look into this with a degugger to see what is going on.
